### PR TITLE
Initialise incremental stake when registering funds.

### PR DESCRIPTION
**Pull request by nc6**
This is needed in order to create the correct snapshot at the end of the
first era.

### CODE REVIEW

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Node.hs [hunk 1]
```diff
@@ -567,9 +567,11 @@ registerInitialFunds initialFunds nes = nes {
           SL._reserves = reserves <-> coin (SL.balance initialFundsUtxo)
         }
 
+    initIncrStake = SL._stakeDistro utxoState
     ledgerState' = ledgerState {
           SL._utxoState = utxoState {
-              SL._utxo = utxo'
+              SL._utxo = utxo',
+              SL._stakeDistro = SL.updateStakeDistribution initIncrStake (SL.UTxO mempty) utxo'
```
Comment (nfrisby) [inline ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Node.hs:574]: Would you add a note (or use a local binding) that makes it clear the `SL.UTxO mempty` is the set of UTxO to delete?

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Node.hs [hunk 2]
```diff
@@ -567,9 +567,11 @@ registerInitialFunds initialFunds nes = nes {
           SL._reserves = reserves <-> coin (SL.balance initialFundsUtxo)
         }
```
Comment (nfrisby) [inline ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Node.hs:569]: Style: some of these new `=` and adjacent old `=` ought to align, I think.

**Review by nfrisby — APPROVED** [review]
Approved, with some minor requests.

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Node.hs (hunk 2 above)
Comment (nc6) [reply ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Node.hs:569]: Aligned!

Comment (nfrisby) [reply ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Node.hs:569]: Line 576 and 577 too please :) `SL._utxo = ` and `SL.stakeDistro =`.

Comment (nfrisby) [reply ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Node.hs:569]: (And the I script has opinions on your long `import` line.)

**Comment by nfrisby** [comment]
Looks like the on-disk ledger state serialization changed? So that's either accidental or you'll need to run the test suite(s?) passing in the `--accept` flag (I think) to update the expected golden result files tracked in the repo.

**Comment by nc6** [comment]
bors merge

**Comment by nc6** [comment]
bors retry

**Pull request merged**